share_app/src/main/java/org/odk/share/activities/WifiActivity.java

```diff
@@ -395,36 +400,26 @@ private void startReceiveTask() {
         receiverService.startDownloading(dstAddress, port);
     }
 
-    @Override
-    public boolean onCreateOptionsMenu(Menu menu) {
-        getMenuInflater().inflate(R.menu.wifi_menu, menu);
-        return super.onCreateOptionsMenu(menu);
+    @OnClick(R.id.bScan)
+    public void scanWifi() {
+        if (wifiManager.isWifiEnabled()) {
+            startScan();
+        } else {
+            Toast.makeText(this, getString(R.string.enable_wifi), Toast.LENGTH_LONG).show();
+        }
     }
 
-    @Override
-    public boolean onOptionsItemSelected(MenuItem item) {
-        switch (item.getItemId()) {
-            case R.id.menu_scan:
-                if (wifiManager.isWifiEnabled()) {
-                    startScan();
-                } else {
-                    Toast.makeText(this, getString(R.string.enable_wifi), Toast.LENGTH_LONG).show();
-                }
-                return true;
-
-            case R.id.menu_qr_code:
-                IntentIntegrator integrator = new IntentIntegrator(this);
-                integrator.setPrompt(getString(R.string.scan_qr_code));
-                integrator.setDesiredBarcodeFormats(integrator.QR_CODE_TYPES);
-                integrator.setCameraId(0);
-                integrator.setOrientationLocked(false);
-
-                integrator.setCaptureActivity(CaptureActivity.class);
-                integrator.setBeepEnabled(true);
-                integrator.initiateScan();
-                return true;
-        }
-        return super.onOptionsItemSelected(item);
+    @OnClick(R.id.bScanQRCode)
+    public void scanQRCode() {
+        IntentIntegrator integrator = new IntentIntegrator(this);
```

Comment:
It would look much cleaner with a chain of calls, wouldn't it?

```
        new IntentIntegrator(this)
                .setPrompt(getString(R.string.scan_qr_code))
                .setDesiredBarcodeFormats(QR_CODE_TYPES)
                .setCameraId(0)
                .setOrientationLocked(false)
                .setCaptureActivity(CaptureActivity.class)
                .setBeepEnabled(true)
                .initiateScan();
```